server/responses.ts

```diff
@@ -1,3 +1,19 @@
+import { MergeExclusive } from 'type-fest'
+
+interface BaseResponseType {
+  status: number
+}
+
+interface NormalResponseType extends BaseResponseType {
+  message: string
+}
+
+interface RawResponseType extends BaseResponseType {
+  rawContentType: string
+}
+
+type ResponseType = MergeExclusive<NormalResponseType, RawResponseType>
```

Comment:
Type is unused; this should be exported. Also consider type-checking `responseList` (perhaps by first declaring a temporary variable and then constructing a mapping type using `keyof`)